ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1337,6 +1337,141 @@ \subsection{Block production}
+make and therefore the genesis window will never ever be invoked. Therefore
+whatever value we use for (P)BFT is irrelevant.
+
+Furthermore, if someone uses this consensus layer to run something different
+that the actual Cardano network, either it doesn't make sense to use (P)BFT
+because they want a decentralized network (so $s$ can be set to $3k/f$) or they
+are only going to run (P)BFT and there block production is controlled by the
+core nodes and there will be no choices to make.
+
+We then decided to use $3k/f$ with the values those variables have in Praos to
+define the genesis window length as up to the current protocol choices ((P)BFT -
+TPraos - Praos). The genesis window length will then be just a parameter in a
+way similar to $k$ and will be fixed during the execution.
+
+\subsection{Future approaches}
+
+There is however a subtlety that has to be made precise here. This choice is
+correct for now and as long as $f$ doesn't change. Currently, the active slots
+coefficient is not a votable parameter so there is no current way to change it.
+However, if it was made votable or even era-dependent, further development will
+be required.
+
+This would lead to the following situation, where the chains transition from era
+$E_1$ to era $E_2$ at the dot:
+
+\begin{center}
+\begin{tikzpicture}[yscale=0.75]
+\draw (0, 0) -- (4, 0) coordinate(i);
+\draw (i) -- ++(1,  1) -- ++(1,0) node{$\bullet$} -- ++(5,0) node[right]{$A$};
+\draw (i) -- ++(1, -1) -- ++(1,0) node{$\bullet$} -- ++(5,0) node[right]{$B$};
+\draw [dashed]
+     (i)
+  -- ++( 0,  2)
+  -- ++( 4,  0) node[pos=0.5,above]{$\overbrace{\hspace{4cm}}^{s_1}$}
+  -- ++( 0, -3.4)
+  -- ++(-4,  0)
+  -- cycle;
+  \draw [dotted]
+     (i)
+  -- ++( 0,  3)
+  -- ++( 5,  0) node[pos=0.5,above]{$\overbrace{\hspace{5cm}}^{s_2}$}
+  -- ++( 0, -5)
+  -- ++(-5,  0)
+  -- cycle;
+\end{tikzpicture}
+\end{center}
+
+In this picture, imagine we are moving from $E_1$ that uses $f_1$ to $E_2$
+that uses $f_2$ where $f_1 > f_2$. The length of the second window would be
+bigger. However, it doesn't make sense to compare the part of the chain after
+the transition point using the genesis window from before the transition point
+and conversely it doesn't make sense to compare the part of the chain before the
+transition point using the genesis window from after the transition point.
+
+It would therefore only be sensible to provide a specific transition function
+that can decide in such situations which chain has to be preferred. Very much
+like other combination functions that are relevant for era transitions, it would
+be required to introduce a new field probably in \lstinline!CanHardFork!:
+
+\begin{lstlisting}
+class (...) => CanHardFork xs where
+  ...
+  hardForkDensitiesComparisons :: InPairs AcrossEraDensities xs
+\end{lstlisting}
+
+where \lstinline!AcrossEraDensities! would be defined in a similar way to
+\lstinline!AcrossEraSelection!:
+
+\begin{lstlisting}
```

Comment:
I wonder how exactly the prefix selection function would have to be changed if we'd introduce variable-length genesis windows. Maybe something to discuss on a call?